src/ffi.rs

```diff
@@ -142,22 +150,379 @@ extern "C" {
+    #[allow(non_snake_case)]
+    pub fn geom_to_WKB_buf(&self, g: &GGeom) -> Option<CVec<u8>> {
+        let mut size = 0;
+        unsafe {
+            let ptr = GEOSGeomToWKB_buf_r(self.ptr, g.as_raw(), &mut size);
+            if ptr.is_null() || size == 0 {
+                None
+            } else {
+                Some(CVec::new(ptr, size as _))
+            }
+        }
+    }
+
+    /// Convert [`GGeom`] from HEX format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, GGeom};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    /// let point_geom = GGeom::new("POINT (2.5 2.5)").expect("Invalid geometry");
+    /// let wkb_buf = context_handle.geom_to_HEX_buf(&point_geom)
+    ///                             .expect("conversion to WKB failed");
+    /// let new_geom = context_handle.geom_from_HEX_buf(wkb_buf.as_ref())
+    ///                              .expect("conversion to WKB failed");
+    /// assert!(point_geom.equals(&new_geom) == Ok(true));
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn geom_from_HEX_buf(&self, hex: &[u8]) -> GeosResult<GGeom> {
+        unsafe {
+            GGeom::new_from_raw(GEOSGeomFromHEX_buf_r(self.ptr, hex.as_ptr(), hex.len()))
+        }
+    }
+
+    /// Convert [`GGeom`] to HEX format.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::{GContextHandle, GGeom};
+    ///
+    /// let context_handle = match GContextHandle::init() {
+    ///     Ok(context_handle) => context_handle,
+    ///     Err(e) => panic!("An error occurred: {}", e),
+    /// };
+    /// let point_geom = GGeom::new("POINT (2.5 2.5)").expect("Invalid geometry");
+    /// let wkb_buf = context_handle.geom_to_HEX_buf(&point_geom)
+    ///                             .expect("conversion to WKB failed");
+    /// let new_geom = context_handle.geom_from_HEX_buf(wkb_buf.as_ref())
+    ///                              .expect("conversion to WKB failed");
+    /// assert!(point_geom.equals(&new_geom) == Ok(true));
+    /// ```
+    #[allow(non_snake_case)]
+    pub fn geom_to_HEX_buf(&self, g: &GGeom) -> Option<CVec<u8>> {
+        let mut size = 0;
+        unsafe {
+            let ptr = GEOSGeomToHEX_buf_r(self.ptr, g.as_raw(), &mut size);
+            if ptr.is_null() || size == 0 {
+                None
+            } else {
+                Some(CVec::new(ptr, size as _))
+            }
+        }
+    }
+}
+
+impl Drop for GContextHandle {
+    fn drop<'a>(&'a mut self) {
+        unsafe { GEOS_finish_r(self.ptr) };
+
+        let previous_ptr = self.notice_message.replace(::std::ptr::null_mut());
+        if !previous_ptr.is_null() {
+            // We free the previous closure.
+            let _callback: Box<Option<Box<dyn Fn(&str) + 'a>>> =
+                unsafe { Box::from_raw(previous_ptr as *mut _) };
+        }
```

Comment:
This will mix badly if we use a C lib that also use geos, no?

Reply:
Unless you call this function on the same object but without the binding, it won't. It's not a global function after all.